java/src/main/java/com/logicalclocks/hsfs/engine/TrainingDatasetEngine.java

```diff
@@ -108,12 +108,14 @@ public void save(TrainingDataset trainingDataset, Dataset<Row> dataset, Map<Stri
     trainingDataset.setLocation(apiTD.getLocation());
     trainingDataset.setVersion(apiTD.getVersion());
     trainingDataset.setId(apiTD.getId());
+    trainingDataset.setStorageConnector(apiTD.getStorageConnector());
 
     // Build write options map
     Map<String, String> writeOptions =
         SparkEngine.getInstance().getWriteOptions(userWriteOptions, trainingDataset.getDataFormat());
 
     SparkEngine.getInstance().write(trainingDataset, dataset, writeOptions, SaveMode.Overwrite);
+    return trainingDataset;
```

Comment:
The APIs should be consistent here between feature groups and training datasets. If you change the training dataset `save()` to return the object itself, then the feature group and on demand feature group classes should do the same.